cpp/oneapi/dal/algo/subgraph_isomorphism/backend/cpu/matching.hpp

```diff
@@ -506,11 +493,12 @@ solution<Cpu> engine_bundle<Cpu>::run() {
         engine_array[index].run_and_wait(gstack, busy_engine_count, false);
     });
 
+    solution<Cpu> aggregated_solution(pattern->get_vertex_count(), allocator_);
     for (std::uint64_t i = 0; i < array_size; i++) {
-        bundle_solutions.add(engine_array[i].get_solution());
+        aggregated_solution.append(engine_array[i].get_solution());
         engine_array[i].~matching_engine();
```

Comment:
In nutshell, the way how `engine_array` memory handled is super unsafe. There is pretty large block of code above that can throw exceptions which lead to memory leaks. Please use RAI whenever it's possible (in this case, it's definitely possible). I'm not asking to fix it now, but it's rather comment for the future. However, formally this code violates SDL.

Reply:
ok, it will be changed when we enable construct with internal allocator